db/agendadb/db.go

```diff
@@ -0,0 +1,217 @@
+		numChoices, err = db.Count(&ChoiceLabeled{})
+		if err != nil {
+			log.Infof("Unable to count choices in existing DB: %v\n", err)
+		}
+	}
+
+	agendaDB := &AgendaDB{
+		sdb:        db,
+		NumAgendas: numAgendas,
+		NumChoices: numChoices,
+	}
+	return agendaDB, err
+}
+
+// Close should be called when you are done with the AgendaDB to close the
+// underlying database.
+func (db *AgendaDB) Close() error {
+	return db.sdb.Close()
+}
+
+// StoreAgenda saves an agenda in the database.
+func (db *AgendaDB) StoreAgenda(agenda *AgendaTagged) error {
+	if db == nil || db.sdb == nil {
+		return fmt.Errorf("AgendaDB not initialized")
+	}
+	return db.sdb.Save(agenda)
+}
+
+// LoadAgenda retrieves an agenda corresponding to the specified unique agenda
+// ID, or nil if it does not exist.
+func (db *AgendaDB) LoadAgenda(agendaID string) (*AgendaTagged, error) {
+	if db == nil || db.sdb == nil {
+		return nil, fmt.Errorf("AgendaDB not initialized")
+	}
+	agenda := new(AgendaTagged)
+	if err := db.sdb.One("Id", agendaID, agenda); err != nil {
+		return nil, err
+	}
+	return agenda, nil
+}
+
+// ListAgendas lists all agendas stored in the database in order of StartTime.
+func (db *AgendaDB) ListAgendas() error {
+	if db == nil || db.sdb == nil {
+		return fmt.Errorf("AgendaDB not initialized")
+	}
+	q := db.sdb.Select().OrderBy("StartTime")
+	i := 0
+	return q.Each(new(AgendaTagged), func(record interface{}) error {
+		a := record.(*AgendaTagged)
+		log.Infof("%d: %s\n", i, a.Id)
+		i++
+		return nil
+	})
+}
+
+// AgendaTagged has the same fields as dcrjson.Agenda, but with the Id field
+// marked as the primary key via the `storm:"id"` tag. Fields tagged for
+// indexing by the DB are: StartTime, ExpireTime, Status, and QuorumProgress.
+type AgendaTagged struct {
+	Id             string           `json:"id" storm:"id"`
+	Description    string           `json:"description"`
+	Mask           uint16           `json:"mask"`
+	StartTime      uint64           `json:"starttime" storm:"index"`
+	ExpireTime     uint64           `json:"expiretime" storm:"index"`
+	Status         string           `json:"status" storm:"index"`
+	QuorumProgress float64          `json:"quorumprogress" storm:"index"`
+	Choices        []dcrjson.Choice `json:"choices"`
+	VoteVersion    uint32           `json:"voteversion"`
+}
+
+// ChoiceLabeled embeds dcrjson.Choice along with the AgendaID for the choice,
+// and a string array suitable for use as a primary key. The AgendaID is tagged
+// as an index for quick lookups based on the agenda.
+type ChoiceLabeled struct {
+	AgendaChoice   [2]string `storm:"id"`
+	AgendaID       string    `json:"agendaid" storm:"index"`
+	dcrjson.Choice `storm:"inline"`
+}
+
+// GetVoteAgendasForVersion is used in getting the agendas using the vote versions
+func GetVoteAgendasForVersion(ver int64, client *rpcclient.Client) (agendas []AgendaTagged) {
+	voteInfo, err := client.GetVoteInfo(uint32(ver))
+	if err == nil {
+		for _, v := range voteInfo.Agendas {
```

Comment:
This makes a copy of this `AgendaTagged`.  It's a largish struct, so I'd range w/ an index and do `v := &voteInfo.Agendas[i]`